web/src/server_events.js

```diff
@@ -264,6 +265,14 @@ export function force_get_events() {
 export function home_view_loaded() {
     waiting_on_homeview_load = false;
     get_events_success([]);
+
+    // If user has the all messages (aka home) view as their default
+    // and current view, call select_id on current selected message
+    // again to trigger mark_read event for selected message.
+    if (!recent_topics_util.is_visible() && message_lists.current === message_lists.home) {
```

Comment:
This call should I think be to `narrow_state.is_message_feed_visible` - it has the same effect today, but when we add an inbox view, the `recent_topics_util` version will become wrong.

Reply:
Done.